presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -423,9 +416,24 @@ else if (insertExistingPartitionsBehavior == InsertExistingPartitionsBehavior.ER
 
         validateSchema(partitionName, schema);
 
-        String fileNameWithExtension = fileName + getFileExtension(conf, outputStorageFormat);
+        String extension = getFileExtension(conf, outputStorageFormat);
+        String targetFileName;
+        if (bucketNumber.isPresent()) {
+            targetFileName = computeBucketedFileName(filePrefix, bucketNumber.getAsInt()) + extension;
+        }
+        else {
+            targetFileName = filePrefix + "_" + randomUUID() + extension;
+        }
+
+        String writeFileName;
+        if (isWritingStagingFilesEnabled(session)) {
+            writeFileName = ".tmp.presto." + filePrefix + "_" + randomUUID() + extension;
```

Comment:
Add a comment stating staging file name are currently totally random and have no relationship with target file name.

This is because today `HiveWriterFactory` doesn't have attempt id thus cannot guarantee file names are different when writing to bucketed tables. Otherwise, we can just do 
```java
    writeFileName = ".tmp.presto." + targetFileName;
```

which looks neat :)

Reply:
It's still somewhat related given we have `filePrefix` ;). 

I don't see a particular reason why we want the connector to be aware of attempt. Ideally the connector only needs to know the attempt that is supposed to be committed, and the clean up other generated tmp files using prefix match with best effort.